explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+
+    $ dig @1.1.1.1 +dnssec -t MX isc.org
+
+    ; <<>> DiG 9.18.28-0ubuntu0.24.04.1-Ubuntu <<>> @1.1.1.1 +dnssec -t MX isc.org
+    ; (1 server found)
+    ;; global options: +cmd
+    ;; Got answer:
+    ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 51256
+    ;; flags: qr rd ra ad; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 1
+
+    ;; OPT PSEUDOSECTION:
+    ; EDNS: version: 0, flags: do; udp: 1232
+    ;; QUESTION SECTION:
+    ;isc.org.                       IN      MX
+
+    ;; ANSWER SECTION:
+    isc.org.                300     IN      MX      5 mx.pao1.isc.org.
+    isc.org.                300     IN      MX      10 mx.ams1.isc.org.
+    isc.org.                300     IN      RRSIG   MX 13 2 300 20241029080338 20241015071431 27566 isc.org. LG/cvFmZ8jLz+CM14foaCtwsyCTwKXfVBZV2jcl2UV8zV79QRLs0YXJ3 sjag1vYCqc+Q5AwUi2DB8L/wZR6EJQ==
+
+    ;; Query time: 199 msec
+    ;; SERVER: 1.1.1.1#53(1.1.1.1) (UDP)
+    ;; WHEN: Tue Oct 22 16:44:33 UTC 2024
+    ;; MSG SIZE  rcvd: 187
+
+
+## Other uses
+
+DNSSEC makes it more attractive and secure to store other types of information in DNS zones. Although it has always been possible to store SRV, TXT and other generic records in DNS, now these can be signed, and can thus be relied upon to be true. A well known initiative that leverages DNSSEC for this purpose is DANE: DNS-based Authentication of Named Entities (RFC 6394, RFC 6698, RFC 7671, RFC 7672, RFC 7673).
+
+For example, imagine if the SSH fingerprints for a host you are logging into for the first time were also stored in DNS, and secured via DNSSEC. Instead of being prompted if you recognize the fingerprint, and wish to proceed, all this verification could happen in the background via DNSSEC and the SSHFP resource record published in the DNS zone for that host. OpenSSH is already capable of this via the configuration option `VerifyHostKeyDNS`.
+
+
+## Where does the DNSSEC validation happen?
+
+DNSSEC validation is the act of fetching the DNS data that was requested, their signatures, and performing the cryptographic validation of that data. Who is responsible for that?
+
+It depends.
+
+Let's analyze the simple scenario of a system on a local network performing a DNS query for a domain.
+
+![Simple DNS](../images/ubuntu-local-recursive-dns.png)
+
+Here we have:
+
+ * An Ubuntu system, like a desktop, configured to use a DNS server in the local network.
+ * A DNS server configured to perform recursive queries on behalf of the clients from the local network.
+
+Let's zoom in a little bit on that Ubuntu system:
+
+![Stub Resolver](../images/ubuntu-stub-resolver.png)
+
+When an application needs to translate a hostname to an IP address, it uses standard glibc calls for that job. That is called a stub resolver, or simply a "dns client". This is a very simple client in the sense that it does not perform recursive queries: it expects to dispatch the DNS query to a recursive DNS server, which will do all the hard work.
```

Comment:
"To translate a hostname into an IP address, applications typically rely on standard glibc functions. This process involves a stub resolver, often referred to as a DNS client. A stub resolver is a simple client that doesn't perform recursive queries itself; instead, it delegates the task to a recursive DNS server, which handles the complex query resolution."